examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -522,22 +529,25 @@ get_classname(
   static char cls_str[4] = {'c', 'l', 's', '\0'};
   bool first_self = *(int32_t*)argname == *(int32_t*)self_str && argname[4] == '\0';
   bool first_cls = *(int32_t*)argname == *(int32_t*)cls_str;
+  if (!first_self && !first_cls) {
+    return result;
+  }
 
   // Read class name from $frame->f_localsplus[0]->ob_type->tp_name.
-  if (first_self || first_cls) {
-    void* tmp;
-    // read f_localsplus[0]:
-    result |= bpf_probe_read_user(&tmp, sizeof(void*), cur_frame + offsets->PyFrameObject.f_localsplus);
-    if (first_self) {
-      // we are working with an instance, first we need to get type
-      result |= bpf_probe_read_user(&tmp, sizeof(void*), tmp + offsets->PyObject.ob_type);
-    }
-    result |= bpf_probe_read_user(&tmp, sizeof(void*), tmp + offsets->PyTypeObject.tp_name);
-    result |= bpf_probe_read_user_str(&symbol->classname, sizeof(symbol->classname), tmp);
+  void* tmp;
+  // read f_localsplus[0]:
+  result |= bpf_probe_read_user(&tmp, sizeof(void*), cur_frame + offsets->PyFrameObject.f_localsplus);
+  if (tmp == NULL) {
+    // self/cls is a cellvar. tough luck :/
```

Comment:
Is this the only possible case? `NULL` is treated by CPython as `UnboundLocalError`. 

Technically it's also possible with such a function:
```
def f():
    self = 5
```
because we only check `co_varnames`.

Can add a check for `co_argcount` if it's easy. If not, just leave a TODO explaining it.

Reply:
Sure, it's possible. But that's really not a case I think we should care about.

Reply:
Drop a comment explaining it

Reply:
clarified a bit